### PR TITLE
[AN-5062] First version: Previews for Soundcloud and Spotify links. 

### PR DESCRIPTION
**Pull request by makingthematrix**
New *PartViews and layouts for both. And since they're very similar - a common abstract class.
#### APK
[Download build #8624](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8624/artifact/build/artifact/wire-dev-PR694-8624.apk)
[Download build #8629](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8629/artifact/build/artifact/wire-dev-PR694-8629.apk)
[Download build #8648](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8648/artifact/build/artifact/wire-dev-PR694-8648.apk)
[Download build #8655](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8655/artifact/build/artifact/wire-dev-PR694-8655.apk)
[Download build #8658](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8658/artifact/build/artifact/wire-dev-PR694-8658.apk)
[Download build #8659](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8659/artifact/build/artifact/wire-dev-PR694-8659.apk)
[Download build #8660](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8660/artifact/build/artifact/wire-dev-PR694-8660.apk)
[Download build #8661](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8661/artifact/build/artifact/wire-dev-PR694-8661.apk)
[Download build #8662](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8662/artifact/build/artifact/wire-dev-PR694-8662.apk)

### CODE REVIEW

**Comment by AndroidBob** [comment]
Can one of the admins verify this patch?

**Comment by deanrobertcook** [comment]
Nice!

**Comment by red-avtovo** [comment]
build? Anyone?

**Comment by JPonte** [comment]
build again

**Comment by deanrobertcook** [comment]
build again

**Comment by deanrobertcook** [comment]
build again

**Comment by olivereitalu** [comment]
- The album art should have a play button
- The previews should be inside a container and align to the content width

**Comment by makingthematrix** [comment]
Build again

**Comment by makingthematrix** [comment]
build again

**Comment by olivereitalu** [comment]
- Rounded corners are missing
- Play button should have a circle (32 40% dark)
- Play button size should be 16

File: app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  )
+
+  registerEphemeral(imageView, imageDrawable)
+  registerEphemeral(playView)
+  registerEphemeral(errorView)
+  registerEphemeral(mediaNameView)
+  registerEphemeral(titleView)
+  registerEphemeral(artistView)
+
+  playView.bringToFront()
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala]: Is this needed? The order of the elements on the xml should define which one's on front.

File: app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    _.artwork.fold2(Signal.empty[ImageSource], id => Signal.const[ImageSource](WireImage(id)))
+  }
+
+  private val imageDrawable = new ImageAssetDrawable(
+    image,
+    background = Some(new ColorDrawable(getColor(R.color.content__youtube__background_color)))
+  )
+
+  registerEphemeral(imageView, imageDrawable)
+  registerEphemeral(playView)
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala]: If the playView is set to invisible when the message expires, why do you need to register it here?

File: app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  content.map(_.tpe).map {
+    case Message.Part.Type.SPOTIFY    => (R.string.mediaplayer__message__onspotify, R.drawable.spotify)
+    case Message.Part.Type.SOUNDCLOUD => (R.string.mediaplayer__message__onsoundcloud, R.drawable.soundcloud)
+    case _ => (-1, -1)
+  }.map {
+    case (nameId, iconId) => ( if(nameId == -1) "" else getString(nameId) , if(iconId == -1) null else getDrawable(iconId) )
+  }{
+    case (name, icon) =>
+      mediaNameView.setText(name)
+      registerEphemeral(iconView, icon)
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala]: Same thing as with the playView.

File: app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  private val imageDrawable = new ImageAssetDrawable(
+    image,
+    background = Some(new ColorDrawable(getColor(R.color.content__youtube__background_color)))
+  )
+
+  registerEphemeral(imageView, imageDrawable)
+  registerEphemeral(playView)
+  registerEphemeral(errorView)
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/messages/parts/SoundMediaPartView.scala]: Same thing as with the playView.

**Pull request merged**